### DISCUSSION TITLE
PlatformIO configuration for CubeCell AB01

### DISCUSSION THREAD

**Discussion by d00m178** [jgromes/RadioLib#1258] [General]
hello

sorry for dumb question but I dont understand why this strange behavior happens:

- creating new PlatformIO project in my VSCode:
  board == Heltec CubeCell-Board (HTCC-AB01)
- copy-paste code from, lets say [ping-pong  example](https://github.com/jgromes/RadioLib/blob/master/examples/SX126x/SX126x_PingPong/SX126x_PingPong.ino)
- press PlatformIO:build button - for sure I receive error: `fatal error: RadioLib.h: No such file or directory`
- edit platformio.ini in this project, add these lines:
```
lib_deps =
  jgromes/RadioLib@^7.0.2
build_src_filter = 
	+<SX126x_Transmit.cpp>
```
- saving file and RadioLib library immediately downloads into ....\my_project\\.pio\libdeps\cubecell_board\ folder
- press Build again and - receive new error:
```
Dependency Graph
|-- RadioLib @ 7.0.2
Error: Nothing to build. Please put your source code files to the 'C:\test\my_project\src' folder
```
- and now most interesting - I commenting lines in platformio.ini:
```
; lib_deps =
;   jgromes/RadioLib@^7.0.2
; build_src_filter = 
; 	+<SX126x_Transmit.cpp>
```
- press Build 3rd time and receiving simple error for function definition:

```
src\my_project.cpp: In function 'void setup()':
src\my_project.cpp:56:29: error: 'setFlag' was not declared in this scope
   56 |   radio.setPacketSentAction(setFlag);
      |                             ^~~~~~~
*** [.pio\build\cubecell_board\src\my_project.cpp.o] Error 1
```
which I can simple fix with define this function like this:

`void setFlag();
`
but only I need put this line before implementation of this functions in the code.

- and after all of above I press Build again and it successfully finished.

so I works with COMMENTED lines in platformio.ini..
why?
where Im wrong and how to configure PlatformIO in correct way?

**Comment by jgromes** [comment]
> so I works with COMMENTED lines in platformio.ini..
> why?

Because you are using platformio incorrectly. You have added a `lib_deps` entry, then ran platformio. It noticed that you don't have RadioLib installed, and promptly downloaded it for you. Does that mean it will delete it if you comment out those lines? No, it will not. It will remain there, hence you will not get a build error.

> Error: Nothing to build. Please put your source code files to the 'C:\test\my_project\src' folder

Well ... is there something in `C:\test\my_project\src` folder? Platformio doesn't seem to think so. Why are you adding that `build_src_filter` entry? I haven't seen it in other platformio projects. Your main file also seems to be called `src\my_project.cpp`, not `src\SX126x_Transmit.cpp`.

But most importantly, none of this is a RadioLib question, it's a question "how do I use platformio with a library" - so please direct that to the platformio forums.

**Comment by d00m178** [comment]
> Your main file also seems to be called src\my_project.cpp, not src\SX126x_Transmit.cpp.

this is the issue. thank you!
